src/ngx_http_lua_directive.c

```diff
@@ -464,8 +466,28 @@ ngx_http_lua_rewrite_by_lua(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
         return NGX_CONF_ERROR;
     }
 
-    if (llcf->rewrite_handler) {
-        return "is duplicate";
+    if (rewrite_handlers == NULL) {
+        rewrite_handlers = ngx_array_create(cf->pool, 1,
+                                      sizeof(ngx_http_lua_phase_handler_t));
+
+        if (rewrite_handlers == NULL) {
+            return NGX_CONF_ERROR;
+        }
+
+        llcf->rewrite_handlers = rewrite_handlers;
+    }
+
+    if (rewrite_handlers->nelts >= NGX_HTTP_LUA_MAX_PHASE_COUNT) {
+        ngx_conf_log_error(NGX_LOG_ERR, cf, 0,
+                           "the number of rewrite_by_lua* directives exceeds %d",
+                           NGX_HTTP_LUA_MAX_PHASE_COUNT);
+        return NGX_CONF_ERROR;
+    }
+
+    phase_handler = ngx_array_push(rewrite_handlers);
+    if (phase_handler == NULL) {
+        ngx_conf_log_error(NGX_LOG_ERR, cf, 0, "no memory");
```

Comment:
Again: we do not log no memory errors on allocation failures on the nginx C land.